GG/src/SDL/SDLGUI.cpp

```diff
@@ -381,8 +380,10 @@ namespace {
                 case GL_FRAMEBUFFER_INCOMPLETE_ATTACHMENT_EXT:
                     return "One of the framebuffer attachments is incomplete.";
                 default:
-                	return "Framebuffer creation failed. Status: " + std::to_string(m_status);
-
+                	std::stringstream ss;
+                    ss << "Framebuffer creation failed. Status: " << m_status;
+					static thread_local std::string sloc = std::move(ss).str();
```

Comment:
This looks rather fishy... I think the clean solution is to return a different string literal for all possible values of `m_status` and not do any string concatenation in local temporaries.

Reply:
sry, but I don't know how to convert from unsigned int directly into const char* without string.

Reply:
You don't convert an `int` to a `const char*`. You extend the `switch` block to return a suitable `"string literal"` for each possible value of `m_status`.

Reply:
The point is you can't do the string concatenation of local temporary variables that will go out of scope when the function returns, as you're returning a `char*` that will point to the location of the result of that concatenation, which will not be a valid pointer after the function returns. Using a `stringstream` or combining `"string literals"` and `std::string`s with `+` doesn't change this, if the result is stored in a local and you can't return the result by value or rvalue reference, which you can't because the function signature is just `const char*`.

Reply:
I suppose other way is to do string concatenation in the exception's constructor and save result in a field.

Reply:
Could work, but more allocation that could throw in an exception constructor might not be ideal...